processing/src/test/java/io/druid/query/ChainedExecutionQueryRunnerTest.java

```diff
@@ -281,17 +284,19 @@ public DyingQueryRunner(CountDownLatch start, CountDownLatch stop)
     @Override
     public Sequence<Integer> run(Query<Integer> query, Map<String, Object> responseContext)
     {
-      hasStarted = true;
-      start.countDown();
-      if (Thread.interrupted()) {
-        interrupted = true;
-        stop.countDown();
-        throw new QueryInterruptedException("I got killed");
+      synchronized (this) { // ensure hasStarted and interrupted are updated simultaneously
+        hasStarted = true;
+        start.countDown();
+        if (Thread.interrupted()) {
+          interrupted = true;
+          stop.countDown();
+          throw new QueryInterruptedException("I got killed");
+        }
       }
 
       // do a lot of work
       try {
-        Thread.sleep(500);
+        Thread.sleep(5000);
```

Comment:
if the test passes, the sleep gets interrupted right away. 500ms might be too short if the build server is slow.